invokeai/backend/stable_diffusion/diffusers_pipeline.py

```diff
@@ -127,11 +123,10 @@ def apply_mask(self, latents: torch.Tensor, t) -> torch.Tensor:
             # some schedulers expect t to be one-dimensional.
             # TODO: file diffusers bug about inconsistency?
             t = einops.repeat(t, "-> batch", batch=batch_size)
-        # Noise shouldn't be re-randomized between steps here. The multistep schedulers
-        # get very confused about what is happening from step to step when we do that.
-        mask_latents = self.scheduler.add_noise(self.mask_latents, self.noise, t)
-        # TODO: Do we need to also apply scheduler.scale_model_input? Or is add_noise appropriately scaled already?
-        # mask_latents = self.scheduler.scale_model_input(mask_latents, t)
+
+        if self.noise is not None:
+            mask_latents = self.scheduler.add_noise(self.mask_latents, self.noise, t)
```

Comment:
Why did `noise` become optional here?

I think I had this always on quite deliberately. If the overall noisiness of the input does not match the level expected for _t_, the model does not handle it well.